eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -308,14 +309,22 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: A
       stay
 
     case Event(c: Peer.OpenChannel, d: ConnectedData) =>
-      val (channel, localParams) = createNewChannel(nodeParams, funder = true, c.fundingSatoshis, origin_opt = Some(sender))
-      c.timeout_opt.map(openTimeout => context.system.scheduler.scheduleOnce(openTimeout.duration, channel, Channel.TickChannelOpenTimeout)(context.dispatcher))
-      val temporaryChannelId = randomBytes32
-      val channelFeeratePerKw = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(target = nodeParams.onChainFeeConf.feeTargets.commitmentBlockTarget)
-      val fundingTxFeeratePerKw = c.fundingTxFeeratePerKw_opt.getOrElse(nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(target = nodeParams.onChainFeeConf.feeTargets.fundingBlockTarget))
-      log.info(s"requesting a new channel with fundingSatoshis=${c.fundingSatoshis}, pushMsat=${c.pushMsat} and fundingFeeratePerByte=${c.fundingTxFeeratePerKw_opt} temporaryChannelId=$temporaryChannelId localParams=$localParams")
-      channel ! INPUT_INIT_FUNDER(temporaryChannelId, c.fundingSatoshis, c.pushMsat, channelFeeratePerKw, fundingTxFeeratePerKw, localParams, d.transport, d.remoteInit, c.channelFlags.getOrElse(nodeParams.channelFlags), ChannelVersion.STANDARD)
-      stay using d.copy(channels = d.channels + (TemporaryChannelId(temporaryChannelId) -> channel))
+      if (c.fundingSatoshis >= Channel.MAX_FUNDING && !Features.hasFeature(nodeParams.features, Wumbo)) {
+        sender ! s"fundingSatoshis=${c.fundingSatoshis} is too big, you must enable large channels support in 'eclair.features' to use funding above ${Channel.MAX_FUNDING} (see eclair.conf)"
```

Comment:
I think that you should not be sending a string here, but instead a `Status.Failure(new IllegalArgumentException("your current string"))`.

It may even be better to properly type that error? If you go that way, it will probably need to correctly override toString and may need some JsonSerializer magic to look nice in the CLI?

Reply:
Sending strings was consistent with the rest of the code in `Peer` but indeed very brittle, commit 457ab1c returns `Status.Failure` with a `RuntimeException` inside. I did not change the JSON serializer/cli because it's already handled by [Throwable serializer](https://github.com/ACINQ/eclair/blob/master/eclair-node/src/main/scala/fr/acinq/eclair/api/JsonSerializers.scala#L192) and the reply from the cli looks like:
```
andrea@andrea-tkp:~/workspace/eclair$ eclair-core/eclair-cli -a localhost:8086 -p foobar open --nodeId=038746cc812250549ba04d5f12d1452aa0c35b48ad37ec17141feb8c407c530409 --fundingSatoshis=50000000
fundingSatoshis=Satoshi(50000000) is too big, you must enable large channels support in 'eclair.features' to use funding above Satoshi(16777216) (see eclair.conf)
```

raw API response:
```json
{
  "error": "fundingSatoshis=Satoshi(400000000) is too big, you must enable large channels support in 'eclair.features' to use funding above Satoshi(16777216) (see eclair.conf)"
}
```

I think we should change other responses too but this should go in a separate PR.

Reply:
Sounds good, we can change that in another PR, it's a broader scope than wumbo.